model/riscv_insts_vext_arith.sail

```diff
@@ -0,0 +1,531 @@
+/* ******************************************************************************* */
+/* This file implements part of the vector extension.                              */
+/* Chapter 11: vector integer arithmetic instructions                              */
+/* Chapter 12: vector fixed-point arithmetic instructions                          */
+/* Chapter 16: vector permutation instructions                                     */
+
+/* ******************************************************************************* */
+
+/* **************************OPIVV(VVTYPE)**************************************** */
+
+union clause ast = VVTYPE : (vvfunct6, bits(1), regidx, regidx, regidx)
+
+mapping encdec_vvfunct6 : vvfunct6 <-> bits(6) = {
+  VV_VADD          <-> 0b000000,
+  VV_VSUB          <-> 0b000010,
+  VV_VMINU         <-> 0b000100,
+  VV_VMIN          <-> 0b000101,
+  VV_VMAXU         <-> 0b000110,
+  VV_VMAX          <-> 0b000111,
+  VV_VAND          <-> 0b001001,
+  VV_VOR           <-> 0b001010,
+  VV_VXOR          <-> 0b001011,
+  VV_VRGATHER      <-> 0b001100,
+  VV_VRGATHEREI16  <-> 0b001110,
+  VV_VSADDU        <-> 0b100000,
+  VV_VSADD         <-> 0b100001,
+  VV_VSSUBU        <-> 0b100010,
+  VV_VSSUB         <-> 0b100011,  
+  VV_VSLL          <-> 0b100101,
+  VV_VSMUL         <-> 0b100111,
+  VV_VSRL          <-> 0b101000,
+  VV_VSRA          <-> 0b101001,
+  VV_VSSRL         <-> 0b101010,
+  VV_VSSRA         <-> 0b101011
+}
+
+mapping clause encdec = VVTYPE(funct6, vm, vs2, vs1, vd)
```

Comment:
These need guards

Reply:
Sorry I didn't catch that. Could you please provide an example?

Reply:
`if hasStdExtV()` or whatever it is on either side. See other extensions' encdec clauses.

Reply:
Guards have been added to vector encdec clauses.